#include "parameters.h"
bool        is_first_frame = true;
bool        runtime_pos_log, pcd_save_en, path_en, extrinsic_est_en = false;
bool        scan_pub_en, scan_body_pub_en;
bool        prop_at_freq_of_imu, check_satu, con_frame, cut_frame;
bool        imu_en, gravity_align, non_station_start;
bool        use_pcd_map_ = true;
bool        use_imu_as_input, space_down_sample, publish_odometry_without_downsample;
double      lidar_end_time = 0.0, first_lidar_time = 0.0, time_con = 0.0;
double      last_timestamp_lidar = -1.0, last_timestamp_imu = -1.0;
double      match_s, satu_acc, satu_gyro, cut_frame_time_interval;
double      imu_time_inte;
double      laser_point_cov, acc_norm;
double      vel_cov, acc_cov_input, gyr_cov_input;
double      gyr_cov_output, acc_cov_output, b_gyr_cov, b_acc_cov;
double      imu_meas_acc_cov, imu_meas_omg_cov; 
double      filter_size_surf_min, filter_size_map_min, fov_deg;
double      cube_len; 
double      time_lag_imu_to_lidar = 0.0;
double      init_x_;
double      init_y_;
double      init_z_;
float       plane_thr;
float       DET_RANGE;
int         pcd_index = 0;
int         init_map_size, con_frame_num;
int         lidar_type, pcd_save_interval;
std::string lid_topic, imu_topic,task;
string      priorPCDMapPath;
vector<double> gravity_init(3, 0.0), gravity(3, 0.0);
vector<double> extrinT(3, 0.0);
vector<double> extrinR(9, 0.0);
shared_ptr<Preprocess> p_pre;